pkg/cmd/migrate.go

```diff
@@ -97,11 +100,17 @@ func migrateRun(cmd *cobra.Command, args []string) error {
 	targetRevision := args[0]
 
 	log.Info().Str("targetRevision", targetRevision).Msg("running migrations")
-	if err := manager.Run(context.Background(), migrationDriver, targetRevision, migrate.LiveRun); err != nil {
+	ctx := cmd.Context()
+	if timeout > 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithDeadline(cmd.Context(), time.Now().Add(timeout))
```

Comment:
There is a `context.WithTimeout` that does this add already.

Reply:
oh! how convenient +1